packages/controls/src/widget_dom_events.ts

```diff
@@ -0,0 +1,276 @@
+    // the listener is attached. This is essentially what layerX and layerY
+    // are supposed to be (and are in chrome) but those event properties have
+    // red box warnings in the MDN documentation that they are not part of any
+    // standard and are not on any standards tracks, so get what we need here.
+    var bounding_rect = view.el.getBoundingClientRect();
+    var y_offset = bounding_rect.top;
+    var x_offset = bounding_rect.left;
+    return {
+        'x': Math.round(event.clientX - x_offset),
+        'y': Math.round(event.clientY - y_offset)
+    }
+}
+
+export
+class DOMListenerModel extends WidgetModel {
+    static serializers = {
+        ...WidgetModel.serializers,
+        source: {deserialize: unpack_models}
+    }
+
+    defaults() {
+        return _.extend(super.defaults(), {
+            _model_name: 'DOMListenerModel',
+            source: null,
+            watched_events: [],
+            ignore_modifier_key_events: false,
+            prevent_default_action: false,
+            _attached_listeners: [],
+            _supported_mouse_events: [],
+            _supported_key_events: [],
+            _modifier_keys: ['Shift', 'Control', 'Alt', 'Meta']
+        });
+    }
+
+    initialize(attributes, options: {model_id: string, comm?: any, widget_manager: any}) {
+        super.initialize(attributes, options);
+        this.on('change:source', this.prepare_source, this)
+        this.on('change:watched_events', this.update_listeners, this)
+        this.prepare_source()
+    }
+
+    key_or_mouse(event_type) {
+        if (_.contains(this.get('_supported_mouse_events'), event_type)) {
+            return 'mouse'
+        }
+        if (_.contains(this.get('_supported_key_events'), event_type)) {
+            return 'keyboard'
+        }
+    }
+
+    _cache_listeners(event_type, view, handler) {
+        // Build up a cache of listeners so they can be removed if the
+        // listener changes source or watched events.
+        this.get('_attached_listeners').push({
+            event: event_type,
+            view: view,
+            func: handler,
+        })
+    }
+
+    prepare_source() {
+        // Watch for changes in the models _view_count, and update
+        // DOM listeners when views are created or destroyed.
+        let previous_model = this.previous('source')
+        this.stopListening(previous_model)
+
+        let current_model = this.get('source')
+        if (current_model.name ==  "DOMWidgetModel") {
+            // We never actually listen to a bare DOMWidgetModel. However,
+            // the InstanceDict trait type initializes to a basic version
+            // of the required class...so effectively, we should treat
+            // this model as undefined.
+            return
+        }
+        if (! (typeof(current_model.get('_view_count')) === "number")) {
+            // Sorry, but we need the view count...
+            current_model.set('_view_count', 0)
```

Comment:
I think that is ok, i don't see this disappearing, unless we get feedback that this idea isn't good. Although for your use case, we may want to have events in the model that are send when views are created and destroyed.